magento/magento2
# Set correct discount package value for tablerate

### Description (*)

This PR fixes the issue for the table rate carrier where the 'package value with discount' in the RateRequest is filled with the package value without the discount.

Currently, when you have a cart rule which sets a discount on the cart, the discount will not be used for the table rate.

Steps to reproduce:
1. Create a table rate based on subtotal, f.e. 5 euro from a subtotal of 0, free shipping from a subtotal of 100 euros.
2. Create a cart rule which sets a discount of 10% on cart, on subtotal only
3. Ensure you have an item in your cart which has the discount, with a value of between 105 euros.
4. The cart rule subtracts 10,5 from 105 - resulting in a total of 94,50. Shipping should be 5 euros as the subtotal now is below 100, but it's 0 because the price value without discount is used.

### Related Pull Requests


### Fixed Issues (if relevant)


### Manual testing scenarios (*)

1. Create a table rate based on subtotal, f.e. 5 euro from a subtotal of 0, free shipping from a subtotal of 100 euros.
2. Create a cart rule which sets a discount of 10% on cart, on subtotal only
3. Ensure you have an item in your cart which has the discount, with a value of between 105 euros.
4. The cart rule subtracts 10,5 from 105 - resulting in a total of 94,50. Shipping costs of 5 euros are applied, order total is 99,50.

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#30169: Set correct discount package value for tablerate
